intelmq/lib/bot.py

```diff
@@ -394,5 +399,56 @@ def __load_harmonization_configuration(self):
                             "Key %s is not valid." % _key)
 
 
+class ParserBot(Bot):
+
+    def parse_csv(self, report):
+        """
+        CSV parser.
+        """
+        raw_report = utils.base64_decode(report.get("raw"))
+        for line in csv.reader(io.StringIO(raw_report)):
+            yield line
+
+    def parse(self, report):
+        """
+        Default parser yielding stripped lines.
+        """
+        for line in utils.base64_decode(report.get("raw")).splitlines():
+            yield line.strip()
+
+    def process(self):
+        self.tempdata = []
+        self.__failed = []
+        report = self.receive_message()
+
+        for line in self.parse(report):
+            if not line:
+                continue
+            try:
+                # filter out None
+                events = list(filter(bool, self.parseline(line, report)))
```

Comment:
Shouldn't we have a default parseline? Even if it is just a `return line`, or even one just with `pass` and a comment saying you should override it on your parsers. That way people reading the code don't have to read the documentation to actually know they should override it.

Also... I'm not 100% sure of the readability of this line. But to be honest I don't have a better idea except for creating a loop which would probably become much slower, and definitely bulkier.